x-pack/plugins/fleet/common/types/rest_spec/agent.ts

```diff
@@ -7,22 +7,17 @@
 
 import type { Agent, AgentAction, NewAgentAction } from '../models';
 
+import type { ListResult, ListWithKuery } from './common';
+
 export interface GetAgentsRequest {
-  query: {
-    page: number;
-    perPage: number;
-    kuery?: string;
+  query: ListWithKuery & {
```

Comment:
There is a slight difference here as `page` and `perPage` are optional in `ListWithKuery`, hopefully not a problem. I feel it's still better to use the common interface.